vendor/github.com/ethereum/go-ethereum/whisper/whisperv5/whisper.go

```diff
@@ -718,6 +735,33 @@ func (wh *Whisper) add(envelope *Envelope) (bool, error) {
 	return true, nil
 }
 
+func (w *Whisper) traceIncomingDelivery(isP2P bool, status message.Status, src NewMessage, env *Envelope, rec *ReceivedMessage, err error) {
+	w.traceDelivery(isP2P, message.IncomingMessage, status, src, env, rec, err)
+}
+
+func (w *Whisper) traceOutgoingDelivery(isP2P bool, status message.Status, src NewMessage, env *Envelope, rec *ReceivedMessage, err error) {
+	w.traceDelivery(isP2P, message.OutgoingMessage, status, src, env, rec, err)
+}
+
+func (w *Whisper) traceDelivery(isP2P bool, dir message.Direction, status message.Status, src NewMessage, env *Envelope, rec *ReceivedMessage, err error) {
+	if w.deliveryServer != nil {
```

Comment:
It's kinda nitpick and matter of style, but I think it's more recommended to check for nil and return:

```go
if w.deliveryServer == nil {
    return
}

// rest of the code without indentation
```

It's cleaner and easier to remove those nil checks in the future if needed, without breaking the indentation of the whole function.